python/tskit/trees.py

```diff
@@ -2338,6 +2340,113 @@ def kc_distance(self, other, lambda_=0.0):
         """
         return self._ll_tree.get_kc_distance(other._ll_tree, lambda_)
 
+    @staticmethod
+    def create_empty(*, span=1, record_provenance=True):
+        """
+        Create an empty tree with no nodes. This function is primarily provided for
+        completeness.
+        """
+        tc = tables.TableCollection(sequence_length=span)
+        if record_provenance:
+            # TODO replace with a version of https://github.com/tskit-dev/tskit/pull/243
+            # TODO also make sure we convert all the arguments so that they are
+            # definitely JSON encodable.
+            parameters = {"command": "create_empty", "TODO": "add parameters"}
+            tc.provenances.add_row(
+                record=json.dumps(provenance.get_provenance_dict(parameters))
+            )
+        return tc.tree_sequence().first()
+
+    @staticmethod
+    def create_star(n, *, span=1, branch_length=1, record_provenance=True):
+        """
+        Create a single :class:<Tree> and associated tree sequence where the ``n``
+        leaf nodes in the tree all have the same parent (i.e. a "star" tree).
+        The leaf nodes are all at time 0 and are marked as sample nodes.
+
+        :param int n: The number of leaf nodes in the tree
+        :param float span: The span of the tree, and therefore the
+            :attr:`~TreeSequence.sequence_length` of the :attr:`~Tree.tree_sequence`
+            property of the returned :class:<Tree>
+        :param float branch_length: The length of every branch in the tree (equivalent
+            to the time of the root node).
+
+        .. note::
+            This is similar to ``tskit.Tree.unrank((0,0), n, span=span)`` but is more
+            efficient for large n. However, the ``unrank`` method provides
+            a concise way of generating alternative (non-star) topologies.
+        """
+        tc = tables.TableCollection(sequence_length=span)
+        tc.nodes.set_columns(
+            flags=np.full(n, NODE_IS_SAMPLE, dtype=np.uint32),
+            time=np.zeros(n),
+        )
+        if n > 1:
```

Comment:
I don't think we should make this exception. We should always return a tree with ``num_leaves`` + 1 nodes and num_leaves edges. (which holds true for all num_leaves >= 0, right?)

Reply:
I was trying to match `unrank`, @jeromekelleher . For num_leaves=1 that produces a TS with a single node and no edges. I think this is the right behaviour. We don't really want a unary node as the root, do we - it would be removed by `simplify()` if we were to ruin that on the TS anyway.

Reply:
Hmm. Maybe we should just raise a ValueError if n < 2 for now, and avoid this discussion. It's not particularly interesting what the semantics are in these degenerate cases and we can always fix them later if it does become important.

Reply:
Yep, could do, I suppose.